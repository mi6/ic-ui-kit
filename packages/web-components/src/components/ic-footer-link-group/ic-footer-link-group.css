@import "../../global/normalize.css";

:host(.footer-link-group-sparse) {
  --footer-link-group-margin-right: 5.938rem;
  --footer-link-group-border-bottom: 0 solid transparent;
  --footer-link-group-title-margin-top: 0;
  --footer-link-group-title-margin-bottom: var(--ic-space-md);
}

:host(.footer-link-group-small) {
  --footer-link-group-margin-right: 0;
  --footer-link-group-title-margin-top: var(--ic-space-md);
  --footer-link-group-title-margin-bottom: var(--ic-space-md);
  --footer-link-group-links-padding-left: var(--ic-space-md);
  --footer-link-group-links-padding-bottom: var(--ic-space-md);
}

:host(.footer-link-group-light.footer-link-group-small) {
  --footer-link-group-border-bottom: var(--ic-keyline-lighten);
}

:host(.footer-link-group-dark.footer-link-group-small) {
  --footer-link-group-border-bottom: var(--ic-keyline-darken);
}

:host(.footer-link-group) {
  margin-right: var(--footer-link-group-margin-right);
  border-bottom: var(--footer-link-group-border-bottom);
}

:host(.footer-link-group-small.footer-link-group-light:hover) {
  background-color: var(--ic-action-light-bg-hover);
  cursor: pointer;
}

:host(.footer-link-group-small.footer-link-group-dark:hover) {
  background-color: var(--ic-action-dark-bg-hover);
  cursor: pointer;
}

:host(.footer-link-group-sparse:focus) {
  box-shadow: var(--ic-border-focus-inset);
  border-radius: var(--ic-border-radius-inset);
  outline: var(--ic-hc-focus-outline);
  z-index: 1;
}

:host(.footer-link-group-small:focus) {
  box-shadow: var(--ic-border-focus-inset);
  border-radius: var(--ic-border-radius-inset);
  outline: var(--ic-hc-focus-outline);
  z-index: 1;
}

:host(.footer-link-group-small.footer-link-group-light:active) {
  background-color: var(--ic-action-light-bg-active);
}

:host(.footer-link-group-small.footer-link-group-dark:active) {
  background-color: var(--ic-action-dark-bg-active);
}

.footer-link-group-title {
  margin-top: var(--footer-link-group-title-margin-top);
  margin-bottom: var(--footer-link-group-title-margin-bottom);
  flex-grow: 1;
}

.footer-link-group-toggle {
  fill: white !important;
}

.footer-link-group-links {
  display: flex;
  flex-direction: column;
}

.footer-link-group-header {
  display: flex;
  align-items: center;
}
