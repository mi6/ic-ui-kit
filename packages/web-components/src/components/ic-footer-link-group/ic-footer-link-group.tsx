import {
  Component,
  Host,
  Element,
  Listen,
  Prop,
  h,
  State,
} from "@stencil/core";
import {
  DEVICE_SIZES,
  getThemeForegroundColor,
  onComponentRequiredPropUndefined,
} from "../../utils/helpers";
import { IcTheme } from "../../utils/types";

@Component({
  tag: "ic-footer-link-group",
  styleUrl: "ic-footer-link-group.css",
  shadow: {
    delegatesFocus: true,
  },
})
export class FooterLinkGroup {
  @Element() el: HTMLIcFooterLinkGroupElement;

  @State() expanded: boolean = false;
  @State() deviceSize: number = DEVICE_SIZES.XL;
  @State() dropdownIconStyle = getThemeForegroundColor();
  @State() small: boolean = false;

  /**
   * The title of the link group to be displayed.
   */
  @Prop() groupTitle!: string;

  componentWillLoad(): void {
    this.small = this.isSmall(this.el);
  }

  componentDidLoad(): void {
    onComponentRequiredPropUndefined(
      [{ prop: this.groupTitle, propName: "group-title" }],
      "Footer Link Group"
    );
  }

  @Listen("footerResized", { target: "document" })
  footerResizeHandler(): void {
    this.small = this.isSmall(this.el);
  }

  @Listen("themeChange", { target: "document" })
  footerThemeChangeHandler(ev: CustomEvent): void {
    const theme: IcTheme = ev.detail;
    this.dropdownIconStyle = theme.mode;
  }

  private isSmall(e: HTMLElement): boolean {
    if (e.parentElement !== null) {
      if (e.parentElement.classList.contains("footer")) {
        return e.parentElement.classList.contains("footer-small");
      } else {
        return this.isSmall(e.parentElement);
      }
    } else {
      return false;
    }
  }

  private handleKeydown = (event: KeyboardEvent): void => {
    if (event.key === " " || event.key === "Enter") {
      this.toggleExpanded();
    }
  };

  private toggleExpanded = (): void => {
    this.expanded = !this.expanded;
  };

  render() {
    const { small, groupTitle } = this;

    return !small ? (
      <Host
        class={{
          ["footer-link-group footer-link-group-sparse"]: true,
          [`footer-link-group-${this.dropdownIconStyle}`]: true,
        }}
      >
        <div class="footer-link-group-title">
          <ic-typography variant="subtitle-small">{groupTitle}</ic-typography>
        </div>
        <div class="footer-link-group-links">
          <slot />
        </div>
      </Host>
    ) : (
      <Host
        class={{
          ["footer-link-group footer-link-group-small"]: true,
          [`footer-link-group-${this.dropdownIconStyle}`]: true,
        }}
        tabIndex="1"
        onClick={this.toggleExpanded}
        onKeydown={this.handleKeydown}
        aria-haspopup="true"
        aria-expanded={this.expanded}
      >
        <ic-section-container fullHeight={true}>
          <div class="footer-link-group-header">
            <div class="footer-link-group-title">
              <ic-typography variant="label">{groupTitle}</ic-typography>
            </div>
            {this.expanded ? (
              <svg
                class="footer-link-group-toggle"
                xmlns="http://www.w3.org/2000/svg"
                aria-hidden="true"
                role="img"
                width="1em"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 1200 1200"
              >
                <path
                  fill="currentColor"
                  d="M600.002 210.605L421.285 389.336L0 810.559l178.721 178.836l421.281-421.341l421.281 421.341L1200 810.559L778.733 389.336L600.002 210.605z"
                />
              </svg>
            ) : (
              <svg
                class="footer-link-group-toggle"
                xmlns="http://www.w3.org/2000/svg"
                aria-hidden="true"
                role="img"
                width="1em"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 1200 1200"
              >
                <g transform="translate(0 1200) scale(1 -1)">
                  <path
                    fill="currentColor"
                    d="M600.002 210.605L421.285 389.336L0 810.559l178.721 178.836l421.281-421.341l421.281 421.341L1200 810.559L778.733 389.336L600.002 210.605z"
                  />
                </g>
              </svg>
            )}
          </div>
          {this.expanded && (
            <div class="footer-link-group-links">
              <slot />
            </div>
          )}
        </ic-section-container>
      </Host>
    );
  }
}
