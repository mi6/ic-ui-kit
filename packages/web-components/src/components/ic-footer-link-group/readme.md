# ic-footer-link-group



<!-- Auto Generated Below -->


## Properties

| Property                  | Attribute     | Description                                  | Type     | Default     |
| ------------------------- | ------------- | -------------------------------------------- | -------- | ----------- |
| `groupTitle` _(required)_ | `group-title` | The title of the link group to be displayed. | `string` | `undefined` |


## Dependencies

### Depends on

- [ic-typography](../ic-typography)
- [ic-section-container](../ic-section-container)

### Graph
```mermaid
graph TD;
  ic-footer-link-group --> ic-typography
  ic-footer-link-group --> ic-section-container
  style ic-footer-link-group fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


