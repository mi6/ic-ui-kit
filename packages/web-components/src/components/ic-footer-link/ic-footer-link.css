@import "../../global/normalize.css";

:host(.footer-link-ungrouped-sparse) {
  --footer-link-margin-left: 0;
  --footer-link-margin-right: 2.5rem;
  --footer-link-margin-bottom: 0;
}

:host(.footer-link-ungrouped-small) {
  --footer-link-margin-left: 0;
  --footer-link-margin-right: 0;
  --footer-link-margin-bottom: var(--ic-space-md);
}

:host(.footer-link-grouped-sparse) {
  --footer-link-margin-left: 0;
  --footer-link-margin-right: 0;
  --footer-link-margin-bottom: var(--ic-space-md);
}

:host(.footer-link-grouped-small) {
  --footer-link-margin-left: var(--ic-space-md);
  --footer-link-margin-right: 0;
  --footer-link-margin-bottom: var(--ic-space-md);
}

:host(.footer-link) {
  margin-left: var(--footer-link-margin-left);
  margin-right: var(--footer-link-margin-right);
  margin-bottom: var(--footer-link-margin-bottom);
}

/* Link styling */

:host(.footer-link) a,
:host(.footer-link) a ::slotted(a) {
  display: flex;
  align-items: flex-end;
  font-weight: var(--ic-font-weight-bold);
  color: inherit;
  width: fit-content;
  font-size: var(--ic-font-size-regular);
}

:host(.footer-link) a:link,
:host(.footer-link) a ::slotted(a:link) {
  color: inherit;
  text-decoration: underline;
  font-weight: var(--ic-font-weight-bold);
}

:host(.footer-link) a:link > ::slotted(svg),
:host(.footer-link) a:link > ::slotted(img),
:host(.footer-link) a ::slotted(a:link > ::slotted(svg)),
:host(.footer-link) a ::slotted(a:link > ::slotted(img)) {
  padding: var(--ic-space-xxxs);
  pointer-events: none;
  transition: var(--ic-easing-transition-fast);
}

/*
There's an apparent defect in Chrome that means this isn't always re-rendered if changed.
Not an issue in Safari. Line 73-79 implement the same function but less clean.

a:link > ::slotted(svg) ,
a:link:visited > ::slotted(svg) {
  fill: var(--ic-theme-text);
}
*/

:host(.footer-link-light) ::slotted(svg) {
  fill: var(--ic-architectural-white);
}

:host(.footer-link-dark) ::slotted(svg) {
  fill: var(--ic-architectural-black);
}

:host(.footer-link) a:link:hover,
:host(.footer-link) a:link:focus,
:host(.footer-link) a ::slotted(a:link:hover),
:host(.footer-link) a ::slotted(a:link:focus) {
  border-bottom: 0.25rem solid !important;
  margin-bottom: -0.25rem !important;
  text-decoration: none;
}

@supports (text-decoration-thickness: 25%) {
  :host(.footer-link) a:link:hover,
  :host(.footer-link) a:link:focus,
  :host(.footer-link) a ::slotted(a:link:hover),
  :host(.footer-link) a ::slotted(a:link:focus) {
    text-decoration-line: underline;
    text-decoration-thickness: 25%;
    text-underline-offset: 25%;
    border-bottom: 0 !important;
    margin-bottom: 0 !important;
  }
}

:host(.footer-link) a:link:hover,
:host(.footer-link) a ::slotted(a:link:hover) {
  outline: none;
}

:host(.footer-link) a:link:focus,
:host(.footer-link) a ::slotted(a:link:focus) {
  outline: var(--ic-hc-focus-outline);
}

:host(.footer-link) a:link:focus > ::slotted(svg),
:host(.footer-link) a:link:focus > ::slotted(img),
:host(.footer-link) a ::slotted(a:link:focus > ::slotted(svg)),
:host(.footer-link) a ::slotted(a:link:focus > ::slotted(img)) {
  box-shadow: var(--ic-border-focus);
  border-radius: var(--ic-border-radius);
}

:host(.footer-link) a:focus,
:host(.footer-link) a ::slotted(a:focus) {
  border-radius: var(--ic-border-radius);
  outline: var(--ic-hc-focus-outline);
  transition: var(--ic-transition-duration-fast);
}

:host(.footer-link) a:link:active,
:host(.footer-link) a:link:focus:active,
:host(.footer-link) a:link:visited:active,
:host(.footer-link) a ::slotted(a:link:active),
:host(.footer-link) a ::slotted(a:link:focus:active),
:host(.footer-link) a ::slotted(a:link:visited:active) {
  color: inherit;
  text-decoration: none;
}

:host(.footer-link) a:hover,
:host(.footer-link) a ::slotted(a:hover) {
  text-decoration: none;
}

.open-in-new-icon {
  padding-left: var(--ic-space-xxs);
}

.open-in-new-icon,
.open-in-new-icon svg {
  height: var(--ic-space-md);
  width: var(--ic-space-md);
}

:host(.footer-link-light) .open-in-new-icon svg {
  fill: var(--ic-architectural-white);
}

:host(.footer-link-dark) .open-in-new-icon svg {
  fill: var(--ic-architectural-black);
}

@media screen and (min-width: 769px) {
  :host(.footer-link.footer-link-ungrouped-sparse:last-of-type) {
    margin-right: 0;
  }
}

@media (forced-colors: active) {
  :host(.footer-link-light) ::slotted(svg),
  :host(.footer-link-dark) ::slotted(svg) {
    fill: currentcolor;
  }
}
