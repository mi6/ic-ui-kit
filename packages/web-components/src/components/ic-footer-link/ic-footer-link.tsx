import {
  Component,
  Element,
  Host,
  Listen,
  Prop,
  h,
  State,
} from "@stencil/core";
import { DEVICE_SIZES, getThemeForegroundColor } from "../../utils/helpers";
import { IcTheme, IcThemeForeground } from "../../utils/types";

import OpenInNew from "../../assets/OpenInNew.svg";

type FooterConfig = { small: boolean; grouped: boolean };

@Component({
  tag: "ic-footer-link",
  styleUrl: "ic-footer-link.css",
  shadow: {
    delegatesFocus: true,
  },
})
export class FooterLink {
  @Element() el: HTMLIcFooterLinkElement;

  @State() deviceSize: number = DEVICE_SIZES.XL;
  @State() footerConfig: FooterConfig = { small: false, grouped: false };
  @State() foregroundColor: IcThemeForeground = getThemeForegroundColor();

  /**
   * If `true`, the user can save the linked URL instead of navigating to it.
   */
  @Prop() download?: string | boolean = false;

  /**
   * The URL that the link points to.
   */
  @Prop() href?: string = null;

  /**
   * The human language of the linked URL.
   */
  @Prop() hreflang?: string;

  /**
   * How much of the referrer to send when following the link.
   */
  @Prop() referrerpolicy?: ReferrerPolicy;

  /**
   * The relationship of the linked URL as space-separated link types.
   */
  @Prop() rel?: string;

  /**
   * The place to display the linked URL, as the name for a browsing context (a tab, window, or iframe).
   */
  @Prop() target?: string;

  componentWillLoad(): void {
    this.footerConfig = this.inferConfig(this.el);
  }

  @Listen("footerResized", { target: "document" })
  footerResizeHandler(): void {
    this.footerConfig = this.inferConfig(this.el);
  }

  @Listen("themeChange", { target: "document" })
  footerThemeChangeHandler(ev: CustomEvent): void {
    const theme: IcTheme = ev.detail;
    this.foregroundColor = theme.mode;
  }

  private inferConfig(e: HTMLElement): FooterConfig {
    if (e.parentElement !== null) {
      if (e.parentElement.classList.contains("footer")) {
        return {
          small: e.parentElement.classList.contains("footer-small"),
          grouped: e.parentElement.classList.contains("footer-grouped"),
        };
      } else {
        return this.inferConfig(e.parentElement);
      }
    } else {
      return { small: false, grouped: false };
    }
  }

  render() {
    const {
      footerConfig,
      href,
      hreflang,
      referrerpolicy,
      rel,
      target,
      download,
    } = this;
    const { small, grouped } = footerConfig;

    return (
      <Host
        class={{
          "footer-link": true,
          [`footer-link-${grouped ? "grouped" : "ungrouped"}-${
            small ? "small" : "sparse"
          }`]: true,
          [`footer-link-${this.foregroundColor}`]: true,
        }}
      >
        <a
          href={href}
          hreflang={hreflang}
          referrerPolicy={referrerpolicy}
          rel={rel}
          download={download !== false ? download : null}
          target={target}
        >
          <slot />
          {target === "_blank" && (
            <span class="open-in-new-icon" innerHTML={OpenInNew} />
          )}
        </a>
      </Host>
    );
  }
}
