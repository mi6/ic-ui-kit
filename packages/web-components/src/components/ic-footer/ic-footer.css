@import "../../global/normalize.css";

:host {
  display: block;

  --footer-compliance-padding: 1rem 0;
}

:host(.footer-sparse) {
  --footer-links-padding: 1.5rem 0;
  --footer-logo-margin-bottom: var(--ic-space-lg);
  --footer-link-inner-flex-direction: row;
}

:host(.footer-small) {
  --footer-links-padding: 0 0;
  --footer-logo-margin-bottom: var(--ic-space-md);
  --footer-link-inner-flex-direction: column;
}

:host(.footer-light) {
  --footer-theme-secondary: var(--ic-theme-secondary);
  --footer-theme-tertiary: var(--ic-theme-tertiary);
  --footer-keyline: var(--ic-keyline-lighten);
}

:host(.footer-dark) {
  --footer-theme-secondary: var(--ic-theme-secondary-light);
  --footer-theme-tertiary: var(--ic-theme-tertiary-light);
  --footer-keyline: var(--ic-keyline-darken);
}

:host(.footer-small.footer-ungrouped) {
  --footer-links-padding: var(--ic-space-md) 0 0 0;
}

/* Main inner footer element */
footer {
  display: flex;
  flex-direction: column;
  width: 100%;
}

/* Description */

.footer-description {
  background-color: var(--footer-theme-secondary);
  color: var(--ic-theme-text);
  border-bottom: var(--footer-keyline);
}

.footer-description-inner {
  padding: 1rem 0;
}

/* Links */

.footer-links {
  padding: var(--footer-links-padding);
  background-color: var(--footer-theme-secondary);
  color: var(--ic-theme-text);
}

.footer-links-inner {
  display: flex;
  flex-direction: var(--footer-link-inner-flex-direction);
}

/* Compliance */

.footer-compliance {
  background-color: var(--footer-theme-tertiary);
  color: var(--ic-theme-text);
}

.footer-compliance-inner {
  padding: var(--footer-compliance-padding);
}

.footer-logo {
  margin-bottom: var(--footer-logo-margin-bottom);
  display: flex;
  gap: var(--ic-space-xxl);
}

.footer-logo > ::slotted() {
  margin-right: var(--ic-space-md);
}

.footer-caption {
  margin-bottom: var(--ic-space-md);
}

.classification-spacing {
  margin-bottom: var(--ic-space-lg);
}

@media (forced-colors: active) {
  footer {
    border-top: var(--ic-hc-border);
  }
}
