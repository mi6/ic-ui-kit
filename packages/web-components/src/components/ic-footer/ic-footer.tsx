import {
  Component,
  Element,
  Event,
  EventEmitter,
  Prop,
  State,
  h,
  Host,
  Listen,
} from "@stencil/core";
import { IC_DEVICE_SIZES } from "../../utils/constants";
import {
  getCurrentDeviceSize,
  getThemeForegroundColor,
  checkResizeObserver,
  hasClassificationBanner,
  isSlotUsed,
} from "../../utils/helpers";
import {
  IcAlignment,
  IcTheme,
  IcThemeForeground,
  IcThemeForegroundEnum,
} from "../../utils/types";
import { IcFooterBreakpoints } from "./ic-footer.types";

/**
 * @slot description - Content will be rendered at the top of the footer.
 * @slot link - Content will be rendered between description and logos.
 * @slot logo - Content will be rendered underneath footer links.
 * @slot caption - Content will be rendered above Crown Copyright.
 */

@Component({
  tag: "ic-footer",
  styleUrl: "ic-footer.css",
  shadow: true,
})
export class Footer {
  private footerEl: HTMLElement;
  private resizeObserver: ResizeObserver;

  @Element() el: HTMLIcFooterElement;

  @State() deviceSize: number = IC_DEVICE_SIZES.XL;
  @State() foregroundColor: IcThemeForeground = getThemeForegroundColor();

  /**
   * The alignment of the section containers used within the footer.
   */
  @Prop() aligned?: IcAlignment = "left";

  /**
   * The screen size breakpoint at which to switch to the small layout.
   */
  @Prop() breakpoint?: IcFooterBreakpoints = "medium";

  /**
   * The caption displayed at the bottom of the footer.
   */
  @Prop() caption: string;

  /**
   * If `true`, the footer will display the crown copyright at the bottom.
   */
  @Prop() copyright: boolean = true;

  /**
   * The description displayed at the top of the footer.
   */
  @Prop() description: string;

  /**
   * If `true`, the footer will be set up to handle link groups instead of standalone links.
   */
  @Prop() groupLinks?: boolean = false;

  /**
   *  @internal Triggers on page resize and triggers style changes in footer links and link groups
   */
  @Event() footerResized: EventEmitter<void>;

  disconnectedCallback(): void {
    this.resizeObserver.disconnect();
  }

  componentWillLoad(): void {
    this.deviceSize = getCurrentDeviceSize();
  }

  componentDidLoad(): void {
    checkResizeObserver(this.runResizeObserver);
  }

  @Listen("themeChange", { target: "document" })
  themeChangeHandler(ev: CustomEvent): void {
    const theme: IcTheme = ev.detail;
    this.foregroundColor = theme.mode;
  }

  private isSmall() {
    const bp = this.breakpoint;

    return bp === "extra small"
      ? this.deviceSize < IC_DEVICE_SIZES.XS
      : bp === "small"
      ? this.deviceSize < IC_DEVICE_SIZES.S
      : bp === "medium"
      ? this.deviceSize < IC_DEVICE_SIZES.M
      : bp === "large"
      ? this.deviceSize < IC_DEVICE_SIZES.L
      : bp === "extra large"
      ? this.deviceSize < IC_DEVICE_SIZES.XL
      : false;
  }

  private resizeObserverCallback = (currSize: number) => {
    if (currSize !== this.deviceSize) {
      this.deviceSize = currSize;
    }
    this.footerResized.emit();
  };

  private runResizeObserver = () => {
    this.resizeObserver = new ResizeObserver(() => {
      const currSize = getCurrentDeviceSize();
      this.resizeObserverCallback(currSize);
    });

    this.resizeObserver.observe(this.footerEl);
  };

  render() {
    const {
      aligned,
      caption,
      copyright,
      description,
      groupLinks,
      foregroundColor,
    } = this;
    const small = this.isSmall();

    return (
      <Host
        class={{
          footer: true,
          [`footer-${small ? "small" : "sparse"}`]: true,
          [`footer-${groupLinks ? "grouped" : "ungrouped"}`]: true,
          [`footer-${foregroundColor}`]: true,
          // Slots will be able to infer their own color
          [IcThemeForegroundEnum.Dark]:
            foregroundColor === IcThemeForegroundEnum.Dark,
          [IcThemeForegroundEnum.Light]:
            foregroundColor === IcThemeForegroundEnum.Light,
        }}
      >
        <footer ref={(footerEl) => (this.footerEl = footerEl)}>
          {/* Description */}
          <div class="footer-description">
            <ic-section-container aligned={aligned} fullHeight={true}>
              <div class="footer-description-inner">
                <ic-typography variant="body">
                  <slot name="description">{description}</slot>
                </ic-typography>
              </div>
            </ic-section-container>
          </div>

          {/* Links */}
          {isSlotUsed(this.el, "link") && (
            <div class="footer-links">
              {groupLinks && small ? (
                <div class="footer-links-inner">
                  <slot name="link" />
                </div>
              ) : (
                <ic-section-container fullHeight aligned={aligned}>
                  <div class="footer-links-inner">
                    <slot name="link" />
                  </div>
                </ic-section-container>
              )}
            </div>
          )}

          {/* Compliance (logo, caption, copyright) */}
          <div class="footer-compliance">
            <ic-section-container aligned={aligned} fullHeight={true}>
              <div class="footer-compliance-inner">
                {isSlotUsed(this.el, "logo") && (
                  <div class="footer-logo">
                    {/* Logo */}
                    <slot name="logo" />
                  </div>
                )}
                {(isSlotUsed(this.el, "caption") || caption) && (
                  <div class="footer-caption">
                    <ic-typography
                      variant={
                        this.deviceSize <= IC_DEVICE_SIZES.M
                          ? "caption"
                          : "body"
                      }
                    >
                      <slot name="caption">{caption}</slot>
                    </ic-typography>
                  </div>
                )}
                {copyright && (
                  <div
                    class={{
                      ["footer-copyright"]: true,
                      ["classification-spacing"]: hasClassificationBanner(),
                    }}
                  >
                    <ic-typography
                      variant={
                        this.deviceSize <= IC_DEVICE_SIZES.M
                          ? "caption-uppercase"
                          : "label-uppercase"
                      }
                    >
                      &copy; Crown Copyright
                    </ic-typography>
                  </div>
                )}
              </div>
            </ic-section-container>
          </div>
        </footer>
      </Host>
    );
  }
}
