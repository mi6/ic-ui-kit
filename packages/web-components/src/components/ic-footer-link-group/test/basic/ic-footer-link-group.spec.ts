import { newSpecPage } from "@stencil/core/testing";
import { Footer } from "../../../ic-footer/ic-footer";
import { FooterLinkGroup } from "../../ic-footer-link-group";
import { FooterLink } from "../../../ic-footer-link/ic-footer-link";

describe("ic-footer-link-group", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [FooterLinkGroup],
      html: `<ic-footer-link-group group-title="Link group"></ic-footer-link-group>`,
    });

    expect(page.root).toEqualHtml(`
    <ic-footer-link-group class="footer-link-group footer-link-group-light footer-link-group-sparse" group-title="Link group">
        <mock:shadow-root>
            <div class="footer-link-group-title">
                <ic-typography variant="subtitle-small">
                    Link group
                </ic-typography>
            </div>
            <div class="footer-link-group-links">
                <slot></slot>
            </div>
        </mock:shadow-root>
    </ic-footer-link-group>`);
  });

  it("should render with links", async () => {
    const page = await newSpecPage({
      components: [FooterLinkGroup, FooterLink],
      html: `<ic-footer-link-group group-title="Link group"><ic-footer-link href="/">Link</ic-footer-link></ic-footer-link-group>`,
    });

    expect(page.root).toEqualHtml(`
    <ic-footer-link-group class="footer-link-group footer-link-group-light footer-link-group-sparse" group-title="Link group">
        <mock:shadow-root>
            <div class="footer-link-group-title">
                <ic-typography variant="subtitle-small">
                    Link group
                </ic-typography>
            </div>
            <div class="footer-link-group-links">
                <slot></slot>
            </div>
        </mock:shadow-root>
        <ic-footer-link class="footer-link footer-link-light footer-link-ungrouped-sparse" href="/">
            <mock:shadow-root>
                <a href="/">
                    <slot></slot>
                </a>
            </mock:shadow-root>
            Link
        </ic-footer-link>
    </ic-footer-link-group>`);
  });

  it("should render within footer", async () => {
    const page = await newSpecPage({
      components: [Footer, FooterLinkGroup, FooterLink],
      html: `<ic-footer><ic-footer-link-group group-title="Link group"><ic-footer-link href="/">Link</ic-footer-link></ic-footer-link-group><ic-footer>`,
    });

    expect(page.root).toMatchSnapshot("footer-link-group-in-footer");
  });

  it("should expand and collapse", async () => {
    const page = await newSpecPage({
      components: [Footer, FooterLinkGroup, FooterLink],
      html: `<ic-footer-link-group group-title="Link group"><ic-footer-link href="/">Link</ic-footer-link></ic-footer-link-group>`,
    });

    expect(page.rootInstance.expanded).toBe(false);

    const event = new KeyboardEvent("keydown", { key: "Enter" });

    await page.rootInstance.handleKeydown(event);

    expect(page.rootInstance.expanded).toBe(true);

    await page.rootInstance.handleKeydown(event);

    expect(page.rootInstance.expanded).toBe(false);
  });

  it("should check and set small state", async () => {
    const page = await newSpecPage({
      components: [Footer, FooterLinkGroup, FooterLink],
      html: `<ic-footer-link-group group-title="Link group"><ic-footer-link href="/">Link</ic-footer-link></ic-footer-link-group>`,
    });

    await page.rootInstance.footerResizeHandler();

    expect(page.rootInstance.small).toBe(false);
  });

  it("should update theme", async () => {
    const page = await newSpecPage({
      components: [Footer, FooterLinkGroup, FooterLink],
      html: `<ic-footer-link-group group-title="Link group"><ic-footer-link href="/">Link</ic-footer-link></ic-footer-link-group>`,
    });

    await page.rootInstance.footerThemeChangeHandler({
      detail: { mode: "dark" },
    });

    expect(page.rootInstance.dropdownIconStyle).toBe("dark");
  });

  it("should render small", async () => {
    const page = await newSpecPage({
      components: [Footer, FooterLinkGroup, FooterLink],
      html: `<ic-footer breakpoint="extra large"><ic-footer-link-group small=true group-title="Link group"><ic-footer-link href="/">Link</ic-footer-link></ic-footer-link-group></ic-footer>`,
    });

    expect(page.root).not.toBeNull;

    expect(page.root).toMatchSnapshot("footer-link-group-small");
  });
});
