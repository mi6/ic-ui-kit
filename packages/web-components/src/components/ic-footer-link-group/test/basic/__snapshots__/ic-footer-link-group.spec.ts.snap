// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-footer-link-group should render small: footer-link-group-small 1`] = `
<ic-footer breakpoint="extra large" class="footer footer-light footer-small footer-ungrouped light">
  <mock:shadow-root>
    <footer>
      <div class="footer-description">
        <ic-section-container aligned="left" fullheight="">
          <div class="footer-description-inner">
            <ic-typography variant="body">
              <slot name="description"></slot>
            </ic-typography>
          </div>
        </ic-section-container>
      </div>
      <div class="footer-compliance">
        <ic-section-container aligned="left" fullheight="">
          <div class="footer-compliance-inner">
            <div class="footer-copyright">
              <ic-typography variant="label-uppercase">
                © Crown Copyright
              </ic-typography>
            </div>
          </div>
        </ic-section-container>
      </div>
    </footer>
  </mock:shadow-root>
  <ic-footer-link-group aria-haspopup="true" class="footer-link-group footer-link-group-light footer-link-group-small" group-title="Link group" small="true" tabindex="1">
    <mock:shadow-root>
      <ic-section-container fullheight="">
        <div class="footer-link-group-header">
          <div class="footer-link-group-title">
            <ic-typography variant="label">
              Link group
            </ic-typography>
          </div>
          <svg aria-hidden="true" class="footer-link-group-toggle" height="1em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 1200 1200" width="1em" xmlns="http://www.w3.org/2000/svg">
            <g transform="translate(0 1200) scale(1 -1)">
              <path d="M600.002 210.605L421.285 389.336L0 810.559l178.721 178.836l421.281-421.341l421.281 421.341L1200 810.559L778.733 389.336L600.002 210.605z" fill="currentColor"></path>
            </g>
          </svg>
        </div>
      </ic-section-container>
    </mock:shadow-root>
    <ic-footer-link class="footer-link footer-link-light footer-link-ungrouped-small" href="/">
      <mock:shadow-root>
        <a href="/">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Link
    </ic-footer-link>
  </ic-footer-link-group>
</ic-footer>
`;

exports[`ic-footer-link-group should render within footer: footer-link-group-in-footer 1`] = `
<ic-footer class="footer footer-light footer-sparse footer-ungrouped light">
  <mock:shadow-root>
    <footer>
      <div class="footer-description">
        <ic-section-container aligned="left" fullheight="">
          <div class="footer-description-inner">
            <ic-typography variant="body">
              <slot name="description"></slot>
            </ic-typography>
          </div>
        </ic-section-container>
      </div>
      <div class="footer-compliance">
        <ic-section-container aligned="left" fullheight="">
          <div class="footer-compliance-inner">
            <div class="footer-copyright">
              <ic-typography variant="label-uppercase">
                © Crown Copyright
              </ic-typography>
            </div>
          </div>
        </ic-section-container>
      </div>
    </footer>
  </mock:shadow-root>
  <ic-footer-link-group class="footer-link-group footer-link-group-light footer-link-group-sparse" group-title="Link group">
    <mock:shadow-root>
      <div class="footer-link-group-title">
        <ic-typography variant="subtitle-small">
          Link group
        </ic-typography>
      </div>
      <div class="footer-link-group-links">
        <slot></slot>
      </div>
    </mock:shadow-root>
    <ic-footer-link class="footer-link footer-link-light footer-link-ungrouped-sparse" href="/">
      <mock:shadow-root>
        <a href="/">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Link
    </ic-footer-link>
  </ic-footer-link-group>
  <ic-footer class="footer footer-light footer-sparse footer-ungrouped light">
    <mock:shadow-root>
      <footer>
        <div class="footer-description">
          <ic-section-container aligned="left" fullheight="">
            <div class="footer-description-inner">
              <ic-typography variant="body">
                <slot name="description"></slot>
              </ic-typography>
            </div>
          </ic-section-container>
        </div>
        <div class="footer-compliance">
          <ic-section-container aligned="left" fullheight="">
            <div class="footer-compliance-inner">
              <div class="footer-copyright">
                <ic-typography variant="label-uppercase">
                  © Crown Copyright
                </ic-typography>
              </div>
            </div>
          </ic-section-container>
        </div>
      </footer>
    </mock:shadow-root>
  </ic-footer>
</ic-footer>
`;
